test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -364,6 +369,26 @@ public static void assertResponse(ActionRequestBuilder<?, SearchResponse> search
         }
     }
 
+    public static void assertScrollResponses(SearchRequestBuilder searchRequestBuilder, Consumer<List<SearchResponse>> consumer) {
+        var timoutSeconds = 30;
```

Comment:
This is an interesting aproach. Could you check if you can use this method with the test in https://github.com/elastic/elasticsearch/pull/102053? 

Maybe the timeout should be an input to the method?

Reply:
FWIW: This method should be able to handle the cases in `TransportTwoNodesSearchIT` which I am trying to work out. WDYT?

Reply:
Thanks @iverase, I'll take a look.  Sure, I can make the timeout an input.

I tried a few different approaches, but settled on making the consumer accept the list of partial responses.  That way, the caller can decide if they want to validate individual aspects of the the response and/or the whole thing.

Reply:
Check out https://github.com/elastic/elasticsearch/pull/102189/commits/0ab8fd6da8675cf1ac7967e72dff6731baa57c80

This updates the test you linked.

I changed `assertScrollResponses` to return a record of "slices" on the result list.  This now makes it easy for the tests to iterate over the the various aspects of the Scroll and perform thier assertions without duplicating the slicing logic each time.